Keyboard smooth scrolling animation is not disabled when `NSScrollAnimationEnabled` is `false`

rr-codes opened the pull request:
#### 207c359c1849663ffceaf90b4d5aae6727eb4230
<pre>
Keyboard smooth scrolling animation is not disabled when `NSScrollAnimationEnabled` is `false`
<a href="https://bugs.webkit.org/show_bug.cgi?id=247822">https://bugs.webkit.org/show_bug.cgi?id=247822</a>
rdar://101225956

Reviewed by Simon Fraser.

Previously, setting `NSScrollAnimationEnabled` to `false` would disable all
scroll animations. However, enabling keyboard smooth scrolling regressed this
and did not respect this setting any longer.

This PR fixes this by having `EventHandlerDrivenSmoothScrolling` respect the
value of `NSScrollAnimationEnabled` so now scroll animations may once again be
disabled.

* Source/WTF/Scripts/Preferences/WebPreferencesInternal.yaml:

Canonical link: <a href="https://commits.webkit.org/256661@main">https://commits.webkit.org/256661@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/cbf021ea57c656b7ec578baf6435b34d3ca8e6ac

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/96373 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/5627 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/29436 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/105915 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/166259 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/100355 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/5792 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/34372 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/88753 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/102643 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/102044 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/4307 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/82973 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/31296 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/86149 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/88041 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/74176 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/87362 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/40104 "Built successfully") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/19536 "Found 1 new test failure: imported/w3c/web-platform-tests/html/canvas/element/manual/imagebitmap/createImageBitmap-drawImage.html (failure)") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/82753 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/37778 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/20922 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/28290 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/4620 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/2541 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/43493 "Passed tests") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/85432 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/515 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/40191 "Passed tests") | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/19262 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on Source/WTF/Scripts/Preferences/WebPreferencesInternal.yaml:
```diff
@@ -289,7 +289,8 @@ EventHandlerDrivenSmoothKeyboardScrollingEnabled:
     WebKit:
-      "PLATFORM(COCOA)": true
+      "PLATFORM(MAC)": WebKit::defaultScrollAnimatorEnabled()
+      "PLATFORM(IOS)": true
```
This should be `PLATFORM(IOS_FAMILY)` since `PLATFORM(COCOA)` is `PLATFORM(MAC) || PLATFORM(IOS_FAMILY)`.

@hortont424 is it expected that `EventHandlerDrivenSmoothKeyboardScrollingEnabled` was true on iOS?

hortont424 replied:
Seems like a surprising choice :) I don't recall how WKKeyboardScrollingAnimator wins....

rr-codes replied:
should I change it to be false by default and only then `WebKit::defaultScrollAnimatorEnabled()` for MAC?

rr-codes committed:
Keyboard smooth scrolling animation is not disabled when `NSScrollAnimationEnabled` is `false`
https://bugs.webkit.org/show_bug.cgi?id=247822
rdar://101225956

Reviewed by Simon Fraser.

Previously, setting `NSScrollAnimationEnabled` to `false` would disable all
scroll animations. However, enabling keyboard smooth scrolling regressed this
and did not respect this setting any longer.

This PR fixes this by having `EventHandlerDrivenSmoothScrolling` respect the
value of `NSScrollAnimationEnabled` so now scroll animations may once again be
disabled.

* Source/WTF/Scripts/Preferences/WebPreferencesInternal.yaml:

Canonical link: https://commits.webkit.org/256661@main
